services/scheduler/src/Scheduler.js

```diff
@@ -0,0 +1,136 @@
+const uuid = require('node-uuid');
+
+const Lib = require('lib');
+const { Flow } = Lib;
+
+async function loop (body, logger, loopInterval) {
+    logger.info('loop TICK');
+    try {
+        await body();
+    } catch (e) {
+        logger.error(e, 'loop body error');
+    }
+    setTimeout(async () => {
+        loop(body, logger, loopInterval);
+    }, loopInterval);    
+}
+
+class Scheduler {
+    constructor(app) {
+        this._logger = app.getLogger().child({service: "Scheduler"});
+        this._crdClient = app.getK8s().getCRDClient();
+        this._channel = app.getAmqpChannel();
+        this._queueCreator = app.getQueueCreator();
+        this._config = app.getConfig();
+        loop(this._loopBody.bind(this), this._logger, 5000);
+    }
+    async _scheduleOne(flow) {
+        this._logger.trace({flowId: flow.id}, 'schedule flow tick');
+         
+        const scheduleRecord = {
+            'taskId': flow.id,
+            'execId': uuid().replace(/-/g, ''),
+            'userId': "DOES NOT MATTER"
+        };
+        const msg = {
+            id: uuid.v1(),
+            attachments: {},
+            body: {},
+            headers: {},
+            metadata: {}
+        };
+        const {
+            messagesQueue,
+            exchangeName,
+            deadLetterRoutingKey
+        } = this._queueCreator.getAmqpStepConfig(flow, flow.getFirstNode().id);
+        await this._queueCreator.assertMessagesQueue(messagesQueue, exchangeName, deadLetterRoutingKey);
+        await this._channel.sendToQueue(
+            this._queueCreator.getAmqpStepConfig(flow, flow.getFirstNode().id).messagesQueue,
+            Buffer.from(JSON.stringify(msg)),
+            {
+                headers:  scheduleRecord
+            }
+        );
+    }
+    async _loopBody() {
+        const flows = (await this._crdClient.flows.get()).body.items;
+        const schedulerRecords = (await this._crdClient.schedulerrecords.get()).body.items;
+        const schedulerRecordsIndex = schedulerRecords.reduce((index, record) => {
+            index[record.metadata.name] = record;
+            return index; 
+        }, {});
+        const now = new Date();
+        for (let flow of flows) {
+            const flowModel = new Flow(flow);
+            const firstStep = flowModel.getFirstNode();
```

Comment:
Naming convention: would be nice to stick with one of the following: `node` or `step`. I would prefer `node`.